File: getting_started/linear_search.py
import random


def search(data, key):
    for i, d in enumerate(data):
        if d == key:
            return i

    return -1


if __name__ == '__main__':
    data = [random.randint(1, 20) for _ in range(20)]
    
    print(data)

    print("input key > ", end='')
    key = int(input())

    result = search(data, key)
    if result != -1:
        print(f"result = {result}")
    else:
        print(f"{key} is not found.")

File: getting_started/linear_search.go
//go:build ignore

package main

import (
	"fmt"
	"math/rand"
)

func createData() []int {
	data := make([]int, 20)

	for i := 0; i < 20; i++ {
		data[i] = rand.Intn(20)
	}

	return data
}

func search(data []int, key int) int {
	for i := 0; i < len(data); i++ {
		if data[i] == key {
			return i
		}
	}

	return -1
}

func main() {
	data := createData()

	fmt.Println(data)

	fmt.Print("input search key > ")
	var key int
	fmt.Scanf("%d", &key)

	result := search(data, key)
	if result != -1 {
		fmt.Printf("result = %d\n", result)
	} else {
		fmt.Printf("%d is not found.", key)
	}
}
